### DISCUSSION TITLE
Update data with State or calling data from new API call setState useEffect Splice

### DISCUSSION THREAD

**Discussion by jimguistical** [nss-evening-cohort-14/evening-client-side#321] [Capstones!] [answered]
Jim Conner
This is the final main feature of my project to hit MVP, and ticket is more rubber duck-y than code-y.
Goal is to allow user to add/remove Service Sites (objects from a nashville.gov API array) so they can update the user's List of Service Sites.

The Q: 
Do I call the specific object I want based on the building's name ("building") from the API, OR should I be using the prevState of my array and manipulating via a new array in State from there?  use splice() to manipulate the array between lists?  Call API for the object vs. moving state around.  
After talking with Dr. T during bookclub though, I understand more clearly that it's better to make another API call for the data I want.  So I think my thinking and my learning have a better perspective now.  For example, I can use useEffect to hide/show cards as they are selected or deleted instead of prop drilling my state down too many levels.  

So the is the direction I'm heading in now, and would appreciate any feedback:

- My first step will be to create a call based on the building name as ID to call the specific  object into the new array and the ListView and update my list Object with the building ID.  I could use a Promise.All to patch my List and push the object into my Array from the API. 
- To remove a SiteCard  from the User's List, I would splice() by building Name as my id to get the card off the ListView.  

- For MVP I don't think I set that exactly as a requirement to update the cards dynamically on SiteCard too, but I need to review my tickets to confirm either way.  So to update state on the ListView can be my main focus right now and update cards first there

Google-foo:
- https://stackoverflow.com/questions/52954908/add-an-object-with-setstate-to-an-array-of-objects
- https://reactjs.org/docs/react-component.html#setstate
- https://stackoverflow.com/questions/39941734/how-can-i-insert-into-reacts-state-array-with-setstate
- https://stackoverflow.com/questions/64026595/how-to-update-state-array-fetched-from-api-in-react-hooks
- https://stackoverflow.com/questions/36326612/delete-item-from-state-array-in-react

Tickets I've reviewed along the way:
#264 #311 #239 #318 #282 

Updated my original  issue ticket for this feature for reference: [UPDATE List of SiteCards as Auth'd User #7](https://github.com/jim-conner/capstone-e14/issues/7)
Updated Data Flow (current ERD screen cap on bottom slide): [Data Flow on Google Slides](https://docs.google.com/presentation/d/1A5SCPPLKYg6TkFqRbmKZ4inmh-UfrmH2fbheuLv-Zm4/edit#slide=id.gdff71a1ef9_0_1)

Code for Ref:
- https://github.com/jim-conner/capstone-e14/blob/update-list/src/App/views/SitesView.js
- https://github.com/jim-conner/capstone-e14/blob/update-list/src/App/components/SiteCard.js
- https://github.com/jim-conner/capstone-e14/blob/update-list/src/helpers/data/siteData.js
- https://github.com/jim-conner/capstone-e14/blob/update-list/src/App/views/ListView.js

**Comment by ajawashington** [comment, marked as answer]
Rubber duck completed!

**Reply by jimguistical** [reply]
Thank you!  I'll come back later and add some context after I work thru the coding.
